File: Model/Site/site.go
package Site

import (
	"fmt"
	db "mmagic/Database"
)

type Site struct {
	db.Model

	SiteTitle     string `json:"site_title" gorm:"type:varchar(190);not null;default '';comment:'网站标题'"`
	SiteDesc      string `json:"site_desc" gorm:"type:text;not null;default '';comment:'网站描述'"`
	SiteKeyboard  string `json:"site_keyboard" gorm:"type:text;not null;default '';comment:'网站关键字'"`
	SiteCopyright string `json:"site_copyright" gorm:"type:varchar(190);not null;default '';comment:'版权'"`
	SiteTel       string `json:"site_tel" gorm:"type:varchar(20);not null;default '';comment:'电话'"`
	SiteEmail     string `json:"site_email" gorm:"type:varchar(50);not null;default '';comment:'邮箱'"`
	SiteAddress   string `json:"site_address" gorm:"type:varchar(100);not null;default '';comment:'地址'"`
	LandLine      string `json:"land_line" gorm:"type:varchar(50);not null;default '';comment:'座机'"`
	ClientTel     string `json:"client_tel" gorm:"type:varchar(50);not null;default '';comment:'400电话'"`
	RecordNumber  string `json:"record_number" gorm:"type:varchar(100);not null;default '';comment:'备案号'"`
}

func GetSite() (site Site) {
	db.Db.First(&site)
	return
}

// @Summer网站信息添加
func AddSite(data map[string]interface{}) bool {
	err := db.Db.Create(&Site{
		SiteTitle:     data["site_title"].(string),
		SiteDesc:      data["site_desc"].(string),
		SiteKeyboard:  data["site_title"].(string),
		SiteCopyright: data["site_copyright"].(string),
		SiteTel:       data["site_tel"].(string),
		SiteEmail:     data["site_email"].(string),
		SiteAddress:   data["site_address"].(string),
		RecordNumber:  data["record_number"].(string),
	})

	if err.Error != nil {
		fmt.Print("基础信息添加失败", err)
		return false
	}
	return true
}

// @Summer 编辑网站信息
func EditSite(id int, data interface{}) bool {
	err := db.Db.Model(&Site{}).Where("id = ?", id).Update(data)
	if err.Error != nil {
		fmt.Print("基础信息编辑失败", err)
		return false
	}
	return true
}

File: Resources/Public/js/site.js
$(".subCon").on("click", function () {
    if ($("#addform #site_title").val() == "") {
        sweetAlert("操作失败","网站标题不能为空",'error');
        return false;
    }
    if ($("#addform #site_desc").val() == "") {
        sweetAlert("操作失败","网站描述不能为空",'error');
        return false;
    }
    if ($("#addform #site_keyboard").val() == "") {
        sweetAlert("操作失败","网站关键字不能为空",'error');
        return false;
    }
    if ($("#addform #site_keyboard").val() == "") {
        sweetAlert("操作失败","网站关键字不能为空",'error');
        return false;
    }
    if ($("#addform #land_line").val() == "" && $("#addform #client_tel").val() == "" && $("#addform #client_tel").val() == "") {
        sweetAlert("操作失败","电话联系方式，必须填写一项",'error');
        return false;
    }
    $.ajax({
        type: "POST",
        dataType: "json",
        url: "/api/v1/addSite",
        data: $("#addform").serialize(),
        success: function (result) {
            if (Number(result.code) == 200) {
                swal({title:"操作成功",type: 'success'},function () {
                    getSiteInfo()
                })
            } else {
                sweetAlert("操作失败",result.msg,'error');
                return false;
            }
        }
    });
    return false
})
getSiteInfo()
function getSiteInfo() {

    $.ajax({
        type: "GET",
        dataType: "json",
        url: "/api/v1/getSite",
        success: function (result) {
            if (Number(result.code) == 200) {
                $("#addform #id").val(result.data.id)
                $("#addform #site_title").val(result.data.site_title)
                $("#addform #site_desc").val(result.data.site_desc)
                $("#addform #site_keyboard").val(result.data.site_keyboard)
                $("#addform #site_copyright").val(result.data.site_copyright)
                $("#addform #site_tel").val(result.data.site_tel)
                $("#addform #land_line").val(result.data.land_line)
                $("#addform #client_tel").val(result.data.client_tel)
                $("#addform #site_email").val(result.data.site_email)
                $("#addform #site_address").val(result.data.site_address)
                $("#addform #record_number").val(result.data.record_number)
            }
        }
    });
}